packages/fds-components/src/components/interactive/Chips.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import DenyIcon from 'fds-icons/lib/react/DenyIcon';
+
+/**
+ * Common pattern where we use href to generate whether
+ * an href or a button is returned, and also allow a master
+ * override where you can specify any element / component as
+ * the root
+ */
+const baseElement = ({ href, rootEl }) => {
+  let Element;
+  if (rootEl) {
+    Element = rootEl;
+  } else if (href) {
+    Element = 'a';
+  } else {
+    Element = 'button';
+  }
+  return Element;
+};
+
+const Chip = ({
+  size,
+  rootEl,
+  theme,
+  isActive,
+  children,
+  label,
+  subtitle,
+  hasClose,
+  ...rest
+}) => {
+  const Element = baseElement({ href: rest.href, rootEl });
+
+  const rootClass = cx('fdsChip', {
+    'fdsChip--active': isActive,
+    'fdsChip--small': size === 'small',
+    'fdsChip--large': size === 'large',
+    'fdsChip--blue': theme === 'blue',
+    'fdsChip--outline': theme === 'outline',
+  });
+  return (
+    <Element {...rest} className={rootClass}>
+      {label}
+      {subtitle && <span className="fdsChip-subtitle">{subtitle}</span>}
+      {hasClose && (
+        <span className="fdsChip-close">
+          <DenyIcon size="xs" />
+        </span>
+      )}
+    </Element>
+  );
+};
+
+Chip.propTypes = {
+  label: PropTypes.string,
+  /**
+   * This allows for overriding of the Chip root element.
+   * Meant to accomodate  `<Link />`
+   */
+  rootEl: PropTypes.func,
```

Comment:
I've seen the prop name `as` used before. Also `component`. `rootEl` might be slightly deceiving since it sounds like it might need to be a dom element or an already rendered react component `rootEl={<div />}` vs `component={'div'}`.

Reply:
yup, that makes sense. I'm down with both of those suggestions. Changing now.

Reply:
I started with component, but felt like it didn't have enough context (sure component, but component rendered where exactly... inside? outside? replacing that base component? etc). `as` I think provides the context I'm looking for.